uxarray/helpers.py

```diff
@@ -3,70 +3,75 @@
 from pathlib import PurePath
 
 
-# helper function to find file type
 def determine_file_type(filepath):
     """Checks file path and contents to determine file type. Supports detection
     of UGrid, SCRIP, Exodus and shape file.
 
-    Parameters: string, required
+    Parameters
+    ----------
+
+    filepath : :class:`string`: required
        Filepath of the file for which the filetype is to be determined.
 
-    Returns: string
-       File type: ug, exo, scrip or shp
+    Returns
+    -------
+    mesh_filetype : :class:`string`:
+        String describing the file type (ug, exo, scrip or shp)
+        to be used by Grid.py
 
     Raises:
        RuntimeError: Invalid file type
     """
     msg = ""
-    # exodus with coord
+    # Extract file name and extension, read into xarray.Dataset to prep
+    path = PurePath(filepath)
+    file_extension = path.suffix
+    # ext_ds = xr.open_dataset(filepath, mask_and_scale=False)
+
+    # Shapefiles (not supported, placeholder)
+    # if file_extension == ".shp":
+    #     mesh_filetype = "shp"
+
+    # NetCDF style files
     try:
-        # extract the file name and extension
-        path = PurePath(filepath)
-        file_extension = path.suffix
-
-        # try to open file with xarray and test for exodus
-        ext_ds = xr.open_dataset(filepath, mask_and_scale=False)["coord"]
-        mesh_filetype = "exo"
-    except KeyError as e:
-        # exodus with coordx
-        try:
-            ext_ds = xr.open_dataset(filepath, mask_and_scale=False)["coordx"]
+        ext_ds = xr.open_dataset(filepath, mask_and_scale=False)
+        if "coord" in ext_ds.data_vars:
             mesh_filetype = "exo"
-        except KeyError as e:
-            # scrip with grid_center_lon
-            try:
-                ext_ds = xr.open_dataset(
-                    filepath, mask_and_scale=False)["grid_center_lon"]
-                mesh_filetype = "scrip"
-            except KeyError as e:
-                # ugrid with Mesh2
-                try:
-                    ext_ds = xr.open_dataset(filepath,
-                                             mask_and_scale=False)["Mesh2"]
-                    mesh_filetype = "ugrid"
-                except KeyError as e:
-                    print("This is not a supported NetCDF file")
-    except (TypeError, AttributeError) as e:
-        msg = str(e) + ': {}'.format(filepath)
-    except (RuntimeError, OSError) as e:
-        # check if this is a shp file
-        # we won't use xarray to load that file
-        if file_extension == ".shp":
-            mesh_filetype = "shp"
-        else:
-            msg = str(e) + ': {}'.format(filepath)
-    except ValueError as e:
-        # check if this is a shp file
-        # we won't use xarray to load that file
+            print("exo")
```

Comment:
Why is the print statement? To me, constructing a "msg", as it was done before, for each case and raising it as an exception/error in except cases or printing it as the report of succesful output sounds like a good way to go.